Explorer/Assets/DCL/EmojiPanel/EmojiSuggestionPanel.cs

```diff
@@ -0,0 +1,54 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine.Pool;
+using Object = UnityEngine.Object;
+
+namespace DCL.Emoji
+{
+    public class EmojiSuggestionPanel
+    {
+        public event Action<EmojiData> OnEmojiSelected;
+
+        private readonly EmojiSuggestionPanelView view;
+        private readonly IObjectPool<EmojiSuggestionView> suggestionItemsPool;
+        private readonly List<EmojiSuggestionView> usedPoolItems = new ();
+
+        public EmojiSuggestionPanel(EmojiSuggestionPanelView view, EmojiSuggestionView emojiSuggestion)
+        {
+            this.view = view;
+
+            suggestionItemsPool = new ObjectPool<EmojiSuggestionView>(
+                () => Object.Instantiate(emojiSuggestion, view.EmojiSuggestionContainer),
+                defaultCapacity: 20,
+                actionOnGet: buttonView => { buttonView.gameObject.SetActive(true); },
+                actionOnRelease: buttonView => { buttonView.gameObject.SetActive(false); }
+            );
+        }
+
+        public void SetPanelVisibility(bool isVisible)
+        {
+            view.gameObject.SetActive(isVisible);
+        }
+
+        public void SetValues(IEnumerable<EmojiData> foundEmojis)
```

Comment:
In order to get count you can use `IReadOnlyList` that will be reusable, you are not obliged to use the least informative contract possible

Reply:
It's regarding the capability of not releasing the whole list, as you will need to know the number of elements